webapp/src/js/reducers/messages.js

```diff
@@ -11,28 +10,37 @@ module.exports = function(state, action) {
   }
 
   switch (action.type) {
-  case actionTypes.ADD_SELECTED_MESSAGE:
-    return Object.assign({}, state, {
-      selected: Object.assign({}, state.selected, {
-        messages: state.selected.messages.concat(action.payload.message)
-      })
-    });
   case actionTypes.CLEAR_SELECTED:
     return Object.assign({}, state, { selected: null });
-  case actionTypes.REMOVE_SELECTED_MESSAGE: {
+  case actionTypes.REMOVE_MESSAGE_FROM_SELECTED_CONVERSATION: {
     const filteredMessages = state.selected.messages.filter(message => message.id !== action.payload.id);
     return Object.assign({}, state, {
       selected: Object.assign({}, state.selected, { messages: filteredMessages })
     });
   }
   case actionTypes.SET_MESSAGES_ERROR:
     return Object.assign({}, state, { error: action.payload.error });
-  case actionTypes.SET_SELECTED_MESSAGE:
+  case actionTypes.SET_SELECTED_CONVERSATION:
     return Object.assign({}, state, { selected: action.payload.selected });
-  case actionTypes.UPDATE_SELECTED_MESSAGE:
+  case actionTypes.UPDATE_SELECTED_CONVERSATION: {
+    const mergedMessages = {};
+    if (action.payload.selected && action.payload.selected.messages) {
+      mergedMessages.messages = ((state.selected && state.selected.messages) || []).map(message => message);
```

Comment:
What does `.map(message => message);` do? Is this meant to be `filter`?

Reply:
I simply generates a new array to hold the elements. Filter would be fine too :) 
Treat at as an `Object.assign`. 
Do you think it's worth adding a comment on top to explain?

Reply:
No it's fine. I think I prefer the spread syntax `[...array]` but it makes sense now. `filter` would be dangerous because it could return an array of a different length.